### PR TITLE
feat(route): cfr

### LINKED ISSUE #15542: 请求新增 对外关系委员会”(Council on Foreign Relations,缩写CFR) 的官方网站为新的订阅源
### Category

Study

### Website URL

https://www.cfr.org/

### Website description

是美国一个专门从事外交政策和国际事务的非营利、无党派的会员制组织、出版商和智库。它被认为是美国最有影响力的外交政策智库。

### What content should be included?

1.  内容一 文章标题
2. 内容二 文章全文输出

### Additional description

_No response_

### This is not a duplicated rss request

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) and [pull requests](https://github.com/DIYgod/RSSHub/pulls) to ensure this rss proposal has not already been requested

### PR DESCRIPTION
**Pull request by KarasuShin**
## Involved Issue / 该 PR 相关 Issue

Close #15542

## Example for the Proposed Route(s) / 路由地址示例

```routes
/cfr/asia/china
/cfr/blog
/cfr/timelines
/cfr/books-reports
/cfr/videos
/cfr/podcasts/world-next-week
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
Missing link to individual item.

File: lib/routes/cfr/utils.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    return {
+        title: linkData?.title ?? $('.article-header-backgrounder__title').text(),
+        pubDate: linkData?.pubDate,
+        description,
+    };
+}
+
+function parsePodcasts($: CheerioAPI): DataItem {
+    const linkData = parseLinkData($);
+    const description = $('.body-content').first().html() ?? '';
```
Comment (TonyRL) [inline lib/routes/cfr/utils.ts:146]: Podcasts should have their audio embeded in either `description` or [`enclosure`](https://docs.rsshub.app/joinus/advanced/advanced-feed#create-a-podcast-feed)

File: lib/routes/cfr/utils.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    }) as Promise<DataItem>;
+}
+
+function parseArticle($: CheerioAPI): DataItem {
+    const linkData = parseLinkData($);
+    let description = parseDescription($('.body-content'), $);
+    const $articleHeader = $('.article-header__image');
+    if ($articleHeader.find('.article-header__image-img').length) {
+        const articleImageSrc = $articleHeader.find('.article-header__image-img').attr('src');
+        description = `<img src="${articleImageSrc}"><br>${description}`;
```
Comment (TonyRL) [inline lib/routes/cfr/utils.ts:64]: Using `figure` from the site is prefered over `img`.

File: lib/routes/cfr/utils.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    };
+}
+
+function parseBlog($: CheerioAPI): DataItem {
+    const linkData = parseLinkData($);
+    const description = parseDescription($('.body-content'), $);
+    return {
+        title: linkData?.title ?? $('.article-header-blog__title').text(),
+        pubDate: linkData?.pubDate,
+        description,
```
Comment (TonyRL) [inline lib/routes/cfr/utils.ts:83]: Please add the header figure as well.

File: lib/routes/cfr/utils.ts [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    let description = parseDescription($('.main-wrapper__article-body .body-content'), $);
+    const summary = $('.main-wrapper__article-body .summary').html();
+    if (summary) {
+        description = `${summary}<br>${description}`;
+    }
+
+    return {
+        title: linkData?.title ?? $('.article-header-backgrounder__title').text(),
+        pubDate: linkData?.pubDate,
+        description,
```
Comment (TonyRL) [inline lib/routes/cfr/utils.ts:140]: Please add the header figure as well.

File: lib/routes/cfr/utils.ts [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+}
+
+function getVideoIframe($ele: Cheerio<Element>) {
+    const setup = $ele.find('video').data('setup') as VideoSetup;
+    if (setup) {
+        const youtubeSource = setup.sources.find((i) => i.type === 'video/youtube');
+        if (youtubeSource) {
+            const videoId = youtubeSource.src.match(/\?v=([^&]+)/)?.[1];
+            if (videoId) {
+                return `<iframe src="https://www.youtube-nocookie.com/embed/${videoId}"></iframe>`;
```
Comment (TonyRL) [inline lib/routes/cfr/utils.ts:225]:
```suggestion
                return `<iframe src="https://www.youtube-nocookie.com/embed/${videoId}" width="640" height="360" frameborder="0" allowfullscreen></iframe>`;
```

**Comment by KarasuShin** [comment]
/test
```routes
/cfr/podcasts/why-it-matters
```

**Comment by KarasuShin** [comment]
route test for `/cfr/podcasts/world-next-week` has been skipped

**Pull request merged**